public/search.js

```diff
@@ -24,14 +24,20 @@ function appendData(data) {
     var search_list = ""
     for (var i = 0; i < data.hits.length; i++) {
         var div = document.createElement("div");
-        filename = data.hits[i].document.name
-        description = data.hits[i].highlights[0].snippet
         abspath = data.hits[i].document.abspath
+        filename = data.hits[i].document.name
+        if (data.hits[i].highlights.hasOwnProperty('snippet')) {
+
```

Comment:
except for this extra unnecessary white line, looks like a fine improvement... to add a sanity test.

but this kind of sanity should be automated, as soon as possible.

not even a function, something deeper. ideally an agpl library, at this point.